File: _1200_MinimumAbsoluteDifference.java
class Solution {
    public List<List<Integer>> minimumAbsDifference(int[] arr) {
        // Key observation: min diff happen between adjacent elements
        // in a sorted array
        Arrays.sort(arr);
        List<List<Integer>> res = new ArrayList();
        int minPair = Integer.MAX_VALUE;
        for (int i = 0; i < arr.length - 1; i++) {
            minPair = Math.min(minPair, arr[i+1] - arr[i]);
        }
        for (int i = 0; i < arr.length - 1; i++) {
            if (arr[i+1] - arr[i] == minPair) {
                res.add(Arrays.asList(arr[i], arr[i+1]));
            }
        }
        return res;
    }
}

File: _1200_MinimumAbsoluteDifference.cpp
#include <iostream>
#include <vector>
#include <algorithm>

class Solution {
public:
    std::vector<std::vector<int>> minimumAbsDifference(std::vector<int>& arr) {
        std::sort(arr.begin(), arr.end());
        std::vector<std::vector<int>> res;
        int minPair = INT_MAX;
        for (int i = 0; i < arr.size() - 1; i++) {
            minPair = std::min(minPair, arr[i+1] - arr[i]);
        }
        for (int i = 0; i < arr.size() - 1; i++) {
            if (arr[i+1] - arr[i] == minPair) {
                res.push_back({arr[i], arr[i+1]});
            }
        }
        return res;
    }
};